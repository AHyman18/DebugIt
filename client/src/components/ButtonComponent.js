import React, { Component } from 'react';

const ButtonComponent = () => {

  return (
    <div className="ButtonComponent">
      ButtonComponent
    </div>
  );
};

export default ButtonComponent;