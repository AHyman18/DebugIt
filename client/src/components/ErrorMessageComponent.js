import React, { Component } from 'react';

const ErrorMessageComponent = () => {

  return (
    <div>
      ErrorMessageComponent
    </div>
  );

};

export default ErrorMessageComponent;