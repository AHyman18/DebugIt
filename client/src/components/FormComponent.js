import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userSignUp, userLogIn } from '../redux/actions/userAction';


class FormComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit(e) {
    e.preventDefault();
    let user = {
      username: this.state.username,
      password: this.state.password
    }
    this.props.signUpAction(user)
  }

  render() {
    return (
      <div className="FormComponent">
        <form onSubmit={this.handleSubmit}>
          <label>
            Username:
          <input
              type="text"
              placeholder="Enter Username"
              name="username"
              value={this.state.value}
              onChange={this.handleChange} />
          </label>
          <label>
            Password:
          <input
              type="password"
              placeholder="Password"
              name="password"
              value={this.state.value}
              onChange={this.handleChange} />
          </label>
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </div >
    )
  }

};

const mapStateToProps = ({ userReducer }) => {
  console.log(userReducer)
  return userReducer;
};

const mapDispatchToProps = (dispatch) => {
  return {
    signUpAction: (user) => dispatch(userSignUp(user)),
    // Dispatch your actions
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(FormComponent);
