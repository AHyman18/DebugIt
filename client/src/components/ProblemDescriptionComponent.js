import React, { Component } from 'react';

const ProblemDescriptionComponent = () => {

  return (
    <div>
      <p>The problem you need to fix is something big</p>

    </div>
  );

};

export default ProblemDescriptionComponent;