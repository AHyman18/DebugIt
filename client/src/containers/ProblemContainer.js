import React, { Component } from 'react';
import CanvasComponent from '../components/CanvasComponent';

const ProblemContainer = () => {

  return (
    <div>
      <h4>ProblemContainer rendered by ProblemPageContainer but is also rendering: </h4>
      <CanvasComponent />
    </div>
  );

};

export default ProblemContainer;