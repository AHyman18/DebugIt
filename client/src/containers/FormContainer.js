import React, { Component } from 'react';
import FormComponent from '../components/FormComponent';

const FormContainer = () => {

  return (
    <div>
      <h4>Form Container rendering:</h4>
      <FormComponent />
    </div>
  );

};

export default FormContainer;