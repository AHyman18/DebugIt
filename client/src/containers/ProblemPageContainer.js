import React, { Component } from 'react';
import ProblemContainer from './ProblemContainer';
import ErrorMessageComponent from '../components/ErrorMessageComponent';

const ProblemPageContainer = () => {

  return (
    <div className="ProblemPageContainer">
      <ErrorMessageComponent />
      <ProblemContainer />
    </div>
  );

};

export default ProblemPageContainer;
