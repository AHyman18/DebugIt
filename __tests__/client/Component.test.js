/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import FormContainer from '../../client/src/containers/FormContainer';
import ButtonComponent from '../../client/src/components/ButtonComponent';
import ErrorMessageComponent from '../../client/src/components/ErrorMessageComponent';
import FormComponent from '../../client/src/components/FormComponent';
import ProblemDescriptionComponent from '../../client/src/components/ProblemDescriptionComponent';
import ProblemContainer from '../../client/src/containers/ProblemContainer';
import ProblemPageContainer from '../../client/src/containers/ProblemPageContainer';


describe('Atomic Components', () => {
  it('Button Component should sucessfully render to the DOM', () => {
    const wrapper = shallow(<ButtonComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Error Message Component should sucessfully render to the DOM', () => {
    const wrapper = shallow(<ErrorMessageComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Form Component should sucessfully render to the DOM', () => {
    const wrapper = shallow(<FormComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Problem Description Component should sucessfully render to the DOM', () => {
    const wrapper = shallow(<ProblemDescriptionComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('Molecular Components', () => {
  it('Form Container should sucessfully render to the DOM', () => {
    const wrapper = shallow(<FormContainer />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Problem Container should successfully render to the DOM', () => {
    const wrapper = shallow(<ProblemContainer />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Problem Page Container should successfully render to the DOM', () => {
    const wrapper = shallow(<ProblemPageContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
