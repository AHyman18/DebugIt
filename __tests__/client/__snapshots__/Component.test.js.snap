// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atomic Components Button Component should sucessfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Atomic Components Error Message Component should sucessfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Atomic Components Form Component should sucessfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Atomic Components Problem Description Component should sucessfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Molecular Components Form Container should sucessfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Molecular Components Problem Container should successfully render to the DOM 1`] = `ShallowWrapper {}`;

exports[`Molecular Components Problem Page Container should successfully render to the DOM 1`] = `ShallowWrapper {}`;
